### PR TITLE
Improve e2e script

### PR DESCRIPTION
**Pull request by johananl**
This PR improves the e2e script.

Summary of changes:

- Avoid altering the user's global kubeconfig file.
- Explicitly use JSON output in all `az` commands to mitigate environment variations.
- Allow overriding script params using env vars.
- Wrap fragile API calls in retry logic.
- Fix masked exit codes in calls to the Azure API.
- Misc cleanups.

Note: This PR depends on https://github.com/Azure/AgentBaker/pull/1310. Therefore, I've opened *this* PR against the 1310 branch as I was told GitHub is smart enough to "redirect" this PR to upstream `master` once the dependency PR is merged :crossed_fingers:

### CODE REVIEW

File: e2e/e2e-script.sh [hunk 1]
```diff
@@ -111,15 +130,16 @@ az vmss create -n ${VMSS_NAME} \
...
 
 # Get the name of the VM instance to later check with kubectl get nodes
 vmInstanceName=$(az vmss list-instances \
                 -n ${VMSS_NAME} \
-                -g $MC_RESOURCE_GROUP_NAME | \
+                -g $MC_RESOURCE_GROUP_NAME \
+                -ojson | \
                 jq -r '.[].osProfile.computerName'
             )
-export vmInstanceName
```
Comment (jepio) [inline e2e/e2e-script.sh:118]: This is consumed by envsubst when generating the pod manifest, so I think the export has to stay.

File: e2e/e2e-script.sh [hunk 2]
```diff
@@ -73,18 +84,28 @@ addJsonToFile "subID" $SUBSCRIPTION_ID
...
+    )
+    retval=$?
+    set -e
+    if [ "$retval" -ne 0 ]; then
+        echo "retrying attempt $i"
+        sleep 10s
+        continue
+    fi
+    break;
+done
```
Comment (jepio) [inline e2e/e2e-script.sh:111]: In the "rare" case that the 10 attempts don't result in success, I think we have to check `retval` after the loop, or else the thing just continues. Had that happen in development once.
A statement such as `[ "$retval" -eq 0 ]` would be enough to assert success.

File: e2e/e2e-script.sh (hunk 1 above)
Comment (johananl) [reply e2e/e2e-script.sh:118]: Yes, found out the hard way :sweat_smile: Thanks.

File: e2e/e2e-script.sh (hunk 2 above)
Comment (johananl) [reply e2e/e2e-script.sh:111]: Good point. Fixing.

**Comment by johananl** [comment]
@jepio feedback is handled. I can't re-request a review for some reason :thinking:

**Comment by jepio** [comment]
I have no buttons to click - ball is in @alexeldeib's court

**Comment by johananl** [comment]
Superseded by https://github.com/Azure/AgentBaker/pull/1344.

**Pull request closed without merging**